### PR TITLE
chore(recipient-warning): 3rd party recipient banner wire up

### PR DESCRIPTION
**Pull request by alfetopito**
# Update 2023-09-15

- Added to swap confirm modal as well 
image

- And limit
image

- And TWAP
image

- Updated text to be shorter and generic

## NOT included
- Styling
- Showing warning in the table row

# Summary

Wiring up banner logic and cleaning up the local storage in favor of a Jotai atom.

# To Test

1. Place SWAP sending to a different recipient
* Activity details should show the warning
image
2. Click on dismiss
* Warning should be gone
3. Refresh the page and open activity details again
* Warning should not be shown

4. Place LIMIT order sending to a different recipient
* Order details should show the warning
image

5. Click on dismiss
* Warning should be gone
6. Refresh the page and open the order details again
* Warning should not be shown

### CODE REVIEW

**Comment by fairlighteth** [comment]
@alfetopito Thanks for reviewing and iterating on this. Looks way cleaner :)

Some issues I found:

- In my PR I made sure we'd only show this warning on pending-like states (waiting for signature, pending, etc.). Not for final states. From this PR:
Screenshot 2023-09-14 at 14 13 40

- In the order receipt (limit orders) when I DISMISS the banner, the warning triangle is not removed. It should be removed together with the banner. It kept on showing even after page refresh (should read from state).
Screenshot 2023-09-14 at 14 14 48

- I'm thinking that perhaps it's good to add that same warning BEFORE you sign an order. This doesn't concern the original use case (malicious signer in a multi-sig) but given we decided to add this warning for all users, I'm thinking it might be good to have the user be aware of this before signing the order intent. WDYT?

Screenshot 2023-09-14 at 14 15 17

**Review by anxolin — COMMENTED** [review]
The styles and how it works is great. 
However, the message for my SWAP can be improved, I already signed the SWAP:
image

Also it says sth about the safe:
image

Also i saw this issue:

image


Are we going to add, maybe in a follow up something more visible here:
image

File: apps/cowswap-frontend/src/common/pure/InlineBanner/banners.tsx [hunk 1]
```diff
@@ -90,22 +89,34 @@ export function SmallVolumeWarningBanner({ feePercentage, feeAmount }: SmallVolu
...
-  if (!isBannerVisible) {
-    return null;
-  }
+export function CustomRecipientWarningBanner({
+  bannerType,
+  borderRadius,
+  orientation,
+  iconSize = 21,
+  iconPadding = '0',
+  padding = '10px 16px',
```
[the remark this answers is not in the document]
Comment (alfetopito) [reply apps/cowswap-frontend/src/common/pure/InlineBanner/banners.tsx:100]: I'll leave that up to @fairlighteth as this is coming from his PR

File: apps/cowswap-frontend/src/legacy/components/swap/SwapModalHeader/SwapModalHeaderMod.tsx
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/components/swap/SwapModalHeader/SwapModalHeaderMod.tsx:1]: Removed mod file

File: apps/cowswap-frontend/src/legacy/components/swap/SwapModalHeader/index.tsx [hunk 2]
```diff
@@ -64,3 +52,200 @@ export default function SwapModalHeader(
...
+  )
+
+  const [exactInLabel, exactOutLabel] = useMemo(() => {
+    return [
+      trade?.tradeType === TradeType.EXACT_OUTPUT ? <Trans>From (incl. fee)</Trans> : null,
+      trade?.tradeType === TradeType.EXACT_INPUT ? <Trans>Receive (incl. fee)</Trans> : null,
+    ]
+  }, [trade])
+
+  const isCustomRecipient = recipient !== null
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/components/swap/SwapModalHeader/index.tsx:108]: Main change in this file. Everything else is moving the mod file.

File: apps/cowswap-frontend/src/legacy/components/swap/SwapModalHeader/index.tsx [hunk 3]
```diff
@@ -64,3 +52,200 @@ export default function SwapModalHeader(
...
+          </ThemedText.Italic>
+        )}
+      </AutoColumn>
+      {/*TODO: adjust styles and maybe merge with the indication below? */}
+      {isCustomRecipient && (
+        <AutoColumn justify="flex-start" gap="sm">
+          <CustomRecipientWarningBanner borderRadius={'12px 12px 0 0'} orientation={BannerOrientation.Horizontal} />
+        </AutoColumn>
+      )}
+      {isCustomRecipient ? (
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/components/swap/SwapModalHeader/index.tsx:235]: Second main change in this file. Everything else is moving the mod file.

File: apps/cowswap-frontend/src/legacy/components/swap/SwapModalHeader/styled.ts
Comment (alfetopito) [inline apps/cowswap-frontend/src/legacy/components/swap/SwapModalHeader/styled.ts:1]: Nothing is new, just refactoring the mod file

File: apps/cowswap-frontend/src/modules/account/containers/Transaction/ActivityDetails.tsx [hunk 4]
```diff
@@ -182,11 +186,8 @@ export function ActivityDetails(props: {
   const { name: receiverEnsName } = useENS(order?.receiver)
 
   // Check if Custom Recipient Warning Banner should be visible
-  const [isCustomRecipientWarningBannerVisible, toggleCustomRecipientWarningBannerVisible] = useBannerVisibility(LSKeys.BANNER_CUSTOM_RECIPIENT_DISMISSED)
-  const [isWarningRecipientIconVisible, setWarningRecipientIconVisible] = useState(isCustomRecipientWarningBannerVisible)
-  useEffect(() => {
-    setWarningRecipientIconVisible(isCustomRecipientWarningBannerVisible)
-  }, [isCustomRecipientWarningBannerVisible])
+  const isCustomRecipientWarningBannerVisible = !useIsReceiverWalletBannerHidden(id) && order && isPending(order)
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/account/containers/Transaction/ActivityDetails.tsx:189]: Only show the banner if order is pending and not yet hidden

File: apps/cowswap-frontend/src/modules/ordersTable/utils/groupOrdersTable.ts
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/ordersTable/utils/groupOrdersTable.ts:1]: Had some app breaking issues, added error handling to not break the page

File: apps/cowswap-frontend/src/modules/twap/hooks/useEmulatedTwapOrders.ts
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/twap/hooks/useEmulatedTwapOrders.ts:1]: Had some app breaking issues, added error handling to not break the page

File: apps/cowswap-frontend/src/modules/twap/utils/emulateTwapAsOrder.ts
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/twap/utils/emulateTwapAsOrder.ts:1]: The types don't allow it, but for some weird reason this was breaking the page, when `executionInfo.info` was undefined.

**Review by fairlighteth — APPROVED** [review]

**Comment by fairlighteth** [comment]
Thanks for this PR. For me good to merge to keep things going.

**Comment by alfetopito** [comment]
Merging as is.
If any comments I'll address them as follow ups

**Pull request merged**